### PR TITLE
Option to report OS memory usage without enabling useGCToTrackMemory

### PR DESCRIPTION
**Pull request by clavin-xlnx**
Signed-off-by: Chris Lavin <chris.lavin@amd.com>

### CODE REVIEW

File: src/com/xilinx/rapidwright/tests/CodePerfTracker.java [hunk 1]
```diff
@@ -280,6 +281,15 @@ public boolean isUsingGCCallsToTrackMemory() {
      */
     public void useGCToTrackMemory(boolean useGCCalls) {
         this.trackMemoryUsingGC = useGCCalls;
+        setTrackOSMemUsage(true);
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/tests/CodePerfTracker.java:284]: Shall we leave a comment that this implies `setTrackOSMemUsage(true)`?
Is it also intentional that even a `useGCToTrackMemory(false)` implies this?

**Review by eddieh-xlnx — COMMENTED** [review]
Shall we update the doc for `setReportingCurrOSMemUsage()` to say it requires `setTrackOSMemUsage(true)` (and enforce it in the code?)

**Comment by eddieh-xlnx** [comment]
I've made a quick change to turn this:
```
==============================================================================
==                                 Example                                  ==
==============================================================================
 XML Parse & Device Load:     1.075s |    655.176MBs (peak)
            (EDIF Parse):            (    0.430s) |    780.953MBs (peak)
    (Read PhysDB Caches):            (    0.078s) |    793.879MBs (peak)
   (Read PhysDB Routing):            (    0.084s) |    830.270MBs (peak)
 (Read PhysDB Placement):            (    0.082s) |    832.297MBs (peak)
(Read PhysDB Finalizing):            (    0.010s) |    832.816MBs (peak)
   Read Netlist & PhysDB:     0.609s |    832.816MBs (peak)
------------------------------------------------------------------------------
         [No GC] *Total*:     1.684s |    832.930MBs (peak)
```

into:
```
==============================================================================
==                                 Example                                  ==
==============================================================================
 XML Parse & Device Load:     1.112s              |    635.684MBs (peak)
            (EDIF Parse):            (    0.447s) |    764.266MBs (peak)
    (Read PhysDB Caches):            (    0.085s) |    788.449MBs (peak)
   (Read PhysDB Routing):            (    0.080s) |    823.824MBs (peak)
 (Read PhysDB Placement):            (    0.082s) |    824.113MBs (peak)
(Read PhysDB Finalizing):            (    0.009s) |    826.758MBs (peak)
   Read Netlist & PhysDB:     0.631s              |    826.758MBs (peak)
------------------------------------------------------------------------------
         [No GC] *Total*:     1.742s              |    826.758MBs (peak)
```
and
```
==============================================================================
==                                 Example                                  ==
==============================================================================
 XML Parse & Device Load:     1.049s              |    620.742MBs (curr) |    620.742MBs (peak)
            (EDIF Parse):            (    0.430s) |    782.520MBs (curr) |    782.520MBs (peak)
    (Read PhysDB Caches):            (    0.082s) |    794.719MBs (curr) |    794.719MBs (peak)
   (Read PhysDB Routing):            (    0.083s) |    830.383MBs (curr) |    830.383MBs (peak)
 (Read PhysDB Placement):            (    0.089s) |    832.906MBs (curr) |    832.906MBs (peak)
(Read PhysDB Finalizing):            (    0.006s) |    833.313MBs (curr) |    833.313MBs (peak)
   Read Netlist & PhysDB:     0.618s              |    833.313MBs (curr) |    833.313MBs (peak)
------------------------------------------------------------------------------
         [No GC] *Total*:     1.667s              |    833.313MBs (curr) |    833.313MBs (peak)
```

File: src/com/xilinx/rapidwright/tests/CodePerfTracker.java [hunk 2]
```diff
@@ -280,6 +285,15 @@ public boolean isUsingGCCallsToTrackMemory() {
...
+    }
+    
+    /**
+     * Sets tracking of OS memory usage (in Linux only).
+     * @param trackOSMemUsage If true, will track curr and peak memory usage of the process as 
+     * reported by the OS. 
+     */
+    public void setTrackOSMemUsage(boolean trackOSMemUsage) {
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/tests/CodePerfTracker.java:296]: Oh, looks like `trackOSMemUsage` is not actually used here!

**Comment by clavin-xlnx** [comment]
Added another fix to adjust output when GC is used.

Before:
```
==============================================================================
==                                 Example                                  ==
==============================================================================
 XML Parse & Device Load:     0.356s     59.643MBs              |    170.684MBs (peak)
      (Read XDEF Header):                      (    0.078s) |    177.168MBs (peak)
   (Read XDEF Placement):                      (    0.040s) |    192.602MBs (peak)
     (Read XDEF Routing):                      (    0.195s) |    267.605MBs (peak)
            (EDIF Parse):                      (    0.607s) |    357.738MBs (peak)
     Read Netlist & XDEF:     0.611s     85.318MBs              |    384.199MBs (peak)
------------------------------------------------------------------------------
                 *Total*:     0.967s    144.961MBs              |    384.199MBs (peak)

```

After:
```
==============================================================================
==                                 Example                                  ==
==============================================================================
 XML Parse & Device Load:     0.362s     59.644MBs          |    173.566MBs (peak)
      (Read XDEF Header):                      (    0.050s) |    173.566MBs (peak)
   (Read XDEF Placement):                      (    0.048s) |    173.566MBs (peak)
     (Read XDEF Routing):                      (    0.178s) |    275.605MBs (peak)
            (EDIF Parse):                      (    0.598s) |    324.457MBs (peak)
     Read Netlist & XDEF:     0.601s     85.310MBs          |    351.777MBs (peak)
------------------------------------------------------------------------------
                 *Total*:     0.963s    144.953MBs          |    351.777MBs (peak)
```

```
==============================================================================
==                                 Example                                  ==
==============================================================================
 XML Parse & Device Load:     0.335s     59.644MBs          |    145.082MBs (curr) |    190.895MBs (peak)
      (Read XDEF Header):                      (    0.060s) |    181.133MBs (curr) |    190.895MBs (peak)
   (Read XDEF Placement):                      (    0.038s) |    194.891MBs (curr) |    194.891MBs (peak)
     (Read XDEF Routing):                      (    0.198s) |    289.625MBs (curr) |    289.625MBs (peak)
            (EDIF Parse):                      (    0.573s) |    343.457MBs (curr) |    343.457MBs (peak)
     Read Netlist & XDEF:     0.576s     85.318MBs          |    335.512MBs (curr) |    357.141MBs (peak)
------------------------------------------------------------------------------
                 *Total*:     0.911s    144.961MBs          |    335.777MBs (curr) |    357.141MBs (peak)
```

File: src/com/xilinx/rapidwright/tests/CodePerfTracker.java (hunk 2 above)
Comment (clavin-xlnx) [reply src/com/xilinx/rapidwright/tests/CodePerfTracker.java:298]: Sorry, that's embarrassing!  Fixed.

**Review by eddieh-xlnx — APPROVED** [review]

**Pull request merged**